### PR TITLE
[gbc] gbc should reject methods with same name

### PR DESCRIPTION
**Pull request by lalo**
- add test
- functions and events should have unique names
- fixes https://github.com/Microsoft/bond/issues/381

### CODE REVIEW

File: compiler/tests/schema/error/duplicate_method_service.bond [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace tests
```
Comment (chwarr) [inline compiler/tests/schema/error/duplicate_method_service.bond]: Minor: consider naming this file `duplicate_service_method.bond`

File: compiler/src/Language/Bond/Parser.hs [hunk 2]
```diff
@@ -240,8 +240,9 @@ struct = do
             [] -> return fields'
             Field {..}:_ -> fail $ "Duplicate definition of the field with ordinal " ++ show fieldOrdinal ++
                 " and name " ++ show fieldName
-      where
-        findDuplicatesBy accessor xs = deleteFirstsBy ((==) `on` accessor) xs (nubBy ((==) `on` accessor) xs)
+
+findDuplicatesBy :: (Eq b) => (a -> b) -> [a] -> [a]
```
Comment (chwarr) [inline compiler/src/Language/Bond/Parser.hs]: Minor: this feels like an odd place in the file for such a helper method. Move to near beginning/end?

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ different versioning scheme, following the Haskell community's
       reader/writer protocols without having to explicitly compute the full
       cross product.
 * Add gbc flags to pick which C# files to generate (structs, gRPC, and comm).
+* gbc ensures that method names are unique [Issue #381](https://github.com/Microsoft/bond/issues/381)
```
Comment (chwarr) [inline CHANGELOG.md]: Minor: "...name are unique with a service."?

Comment (lalo) [inline CHANGELOG.md]: within?

File: CHANGELOG.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ different versioning scheme, following the Haskell community's
       reader/writer protocols without having to explicitly compute the full
       cross product.
 * Add gbc flags to pick which C# files to generate (structs, gRPC, and comm).
-* gbc ensures that method names are unique [Issue #381](https://github.com/Microsoft/bond/issues/381)
+* gbc ensures that method names are unique within a service [Issue #381](https://github.com/Microsoft/bond/issues/381).
```
Comment (chwarr) [inline CHANGELOG.md]: Minor: period before issue link

And yes, I meant "within"

**Pull request closed without merging**